recipe/patches/0004-patch-zoneinfo_dir_override-to-point-to-our-tzdata.patch

```diff
@@ -0,0 +1,111 @@
+From ab018339345371fd38209abd15dd39be024f3f60 Mon Sep 17 00:00:00 2001
+From: "H. Vetinari" <h.vetinari@gmx.com>
+Date: Fri, 21 Jun 2024 12:41:07 +1100
+Subject: [PATCH 4/6] patch zoneinfo_dir_override to point to our tzdata
+
+---
+ libstdc++-v3/src/c++20/tzdb.cc | 76 ++++++++++++++++++++++++++++------
+ 1 file changed, 63 insertions(+), 13 deletions(-)
+
+diff --git a/libstdc++-v3/src/c++20/tzdb.cc b/libstdc++-v3/src/c++20/tzdb.cc
+index c7c7cc9de..5389a9c12 100644
+--- a/libstdc++-v3/src/c++20/tzdb.cc
++++ b/libstdc++-v3/src/c++20/tzdb.cc
+@@ -37,8 +37,12 @@
+ #include <mutex>      // mutex
+ #include <filesystem> // filesystem::read_symlink
+ 
+-#ifndef _AIX
+-# include <cstdlib>   // getenv
++#if defined(__linux__)
++# include <dlfcn.h>   // dladdr
++# include <cstdlib>   // free, malloc, realpath
++# include <cstring>   // memcpy, strlen
++#else defined(_WIN32)
++# include <windows.h>
+ #endif
+ 
+ #if defined __GTHREADS && ATOMIC_POINTER_LOCK_FREE == 2
+@@ -64,25 +68,71 @@
+ 
+ namespace __gnu_cxx
+ {
+-#ifdef _AIX
+-  // Cannot override weak symbols on AIX.
+-  const char* (*zoneinfo_dir_override)() = nullptr;
+-#else
+-  [[gnu::weak]] const char* zoneinfo_dir_override();
+-
+-#if defined(__APPLE__) || defined(__hpux__) \
+-  || (defined(__VXWORKS__) && !defined(__RTP__))
+-  // Need a weak definition for Mach-O et al.
+   [[gnu::weak]] const char* zoneinfo_dir_override()
+   {
++    // get path to library we're in, to determine our location relative to $PREFIX;
```

Comment:
Can we disable this for the static `libstdc++`? There should probably be a preprocessor directive that we can use for it.
When building executables with `-static-libstdc++` like micromamba, we shouldn't leave open a security hole like this.

Reply:
> There should probably be a preprocessor directive that we can use for it.

OK, I'll try to look around.